tests/test_analysis_endpoints.py

```
@@ -0,0 +1,165 @@
+
+# -----------------------------
+# Исключения AnalysisManager → 502
+# -----------------------------
+@pytest.mark.parametrize("endpoint,payload", [
+    ("/v1/analysis/cluster", {"dataset_id": _prepare_dataset("ds-cl-err"), "method": "leiden"}),
+    ("/v1/analysis/metric", {"dataset_id": _prepare_dataset("ds-m-err"), "metric": "pagerank"}),
+])
+def test_analysis_manager_exception(monkeypatch, endpoint, payload):
+    class BadManager:
+        def __init__(self, *a, **k): pass
+        def process(self, ctx): raise RuntimeError("boom")
```

Ruff (0.14.7)

Unused method argument: `ctx`

(ARG002)